### PR TITLE
Remove legacy hints module, rewrite the hints to the newer module

### PR DESCRIPTION
**Pull request by tonihele**
Phew, renewed the hints. I didn't do the unit tests. I tried to, but I get some compilation problems with them and the test wont work. This should be looked into in the future. This is anyway kind of major deprecation removal.

https://platform.netbeans.org/tutorials/nbm-java-hint.html

If you just do them by the instructions, a unit test is added. Looks totally worth writing, as the manual testing is rather tedious.

Also the ReadOnly hint was broken I think. It didn't trigger on "set" method as it should. However, both ReadOnly and Internal are not in use in JME currently (Internal is but not on a method). I tested them with custom classes.

### CODE REVIEW

**Comment by tonihele** [comment]
Test class

```
package mygame;

import com.jme3.app.SimpleApplication;
import com.jme3.material.Material;
import com.jme3.math.ColorRGBA;
import com.jme3.math.Vector3f;
import com.jme3.renderer.RenderManager;
import com.jme3.scene.Geometry;
import com.jme3.scene.Node;
import com.jme3.scene.shape.Box;
import com.jme3.system.Annotations;
import com.jme3.util.BufferAllocatorFactory;
import com.jme3.util.TempVars;
import javax.swing.JOptionPane;

/**
 * This is the Main Class of your Game. You should only do initialization here.
 * Move your Logic into AppStates or Controls
 * @author normenhansen
 */
public class Main extends SimpleApplication {

    public static void main(String[] args) {
        Main app = new Main();
        app.start();
    }

    @Override
    public void simpleInitApp() {
        Box b = new Box(1, 1, 1);
        Geometry geom = new Geometry("Box", b);

        Material mat = new Material(assetManager, "Common/MatDefs/Misc/Unshaded.j3md");
        mat.setColor("Color", ColorRGBA.Blue);
        geom.setMaterial(mat);

        rootNode.updateModelBound();
        TempVars derp = TempVars.get();

        TempVars derpy = TempVars.get();
                JOptionPane.showMessageDialog(null, null);
               // com.jme3.scene.Node.updateGeometricState();
rootNode.getLocalTranslation().set(Vector3f.ZERO);
        String derpf = BufferAllocatorFactory.PROPERTY_BUFFER_ALLOCATOR_IMPLEMENTATION;

        ReadOnlyNode ghfgh = new ReadOnlyNode();
        ghfgh.set(null);
        ghfgh.setKey(null);

        ghfgh.get();
        rootNode.attachChild(geom);
    }



    @Override
    public void simpleUpdate(float tpf) {
        //TODO: add update code
    }

    @Override
    public void simpleRender(RenderManager rm) {
        //TODO: add render code
    }

    @Annotations.ReadOnly
    protected class ReadOnlyNode extends Node {

        @Annotations.ReadOnly
        public void set(String string) {

        }

        @Annotations.Internal
        public void get() {

        }

    }
}

```

**Comment by grizeldi** [comment]
This fixes #37 if I understand correctly?

**Comment by tonihele** [comment]
It looks to me that it very much does solve the issue. Finally.

File: jme3-code-check/src/com/jme3/gde/codecheck/hints/ReadOnlyPrimitiveHint.java [hunk 1]
```diff
@@ -1,144 +1,91 @@
...
-            if (fail) {
-                return Collections.<ErrorDescription>singletonList(
-                        ErrorDescriptionFactory.createErrorDescription(
-                        getSeverity().toEditorSeverity(),
-                        getDisplayName(),
-                        NO_FIXES,
-                        info.getFileObject(),
-                        (int) info.getTrees().getSourcePositions().getStartPosition(info.getCompilationUnit(), t),
-                        (int) info.getTrees().getSourcePositions().getEndPosition(info.getCompilationUnit(), t)));
+            if ("set".equals(el.getSimpleName().toString())) {
```
Comment (tonihele) [inline jme3-code-check/src/com/jme3/gde/codecheck/hints/ReadOnlyPrimitiveHint.java]: This is wrong, it should be startsWith. Like the original tried to do, I assume. I'll fix it after I'm back from work.
```suggestion
            if (el.getSimpleName().toString().startsWith("set")) {
```

File: jme3-code-check/src/com/jme3/gde/codecheck/hints/InternalMethodHint.java [hunk 2]
```diff
@@ -1,134 +1,84 @@
 package com.jme3.gde.codecheck.hints;
 
-import com.jme3.system.Annotations.Internal;
```
Comment (MeFisto94) [inline jme3-code-check/src/com/jme3/gde/codecheck/hints/InternalMethodHint.java]: I think we should open an issue on the engine to check the source and add a few of these where appropriate, as I think noone knows about them anymore :D

**Comment by tonihele** [comment]
There, good to go!

As I said, does what it used to. Except ReadOnlyPrimitiveHint was also fixed as the old failed to check the "set", it looked for it from the wrong place.

**Pull request merged**